### PR TITLE
Fixes: #3950 "Create and Add Another" does retain device type

### LINKED ISSUE #3950: Device "Create and Add Another" does not retain Manufacturer and Device Type
### Environment
* Python version: 3.5.2
* NetBox version:  2.7.1

### Steps to Reproduce
1. Add Device
2. Enter Name, Role, Manufacturer, Device Type, Site and Rack (optionally Tags and Comments)
3. Click "Create and Add Another"

### Expected Behavior

Given the new feature in 2.7, I was expecting the fields to be carried forward into the next form.

### Observed Behavior

Manufacturer and Device Type are *not* carried forward (neither are Comments)

However: Role, Site, Rack and Tags *are* carried forward.

### PR DESCRIPTION
**Pull request by kobayashi**
### Fixes: #3950 "Create and Add Another" does retain device type

Add lines to retrieve manufacture pk while `device_type` is provided.

### CODE REVIEW

**Comment by kobayashi** [comment]
found more abstract approach. I will commit it later again.

**Review by jeremystretch — CHANGES_REQUESTED** [review]

File: netbox/utilities/utils.py [hunk 1]
```diff
@@ -181,15 +181,31 @@ def render_jinja2(template_code, context):
...
+    """
+    if '__' in field_name:
+        fn = field_name.split('__')
+        attr = getattr(instance, fn[0])
+        return get_field_value(attr, fn[1])
+
+    field = instance._meta.get_field(field_name) if instance else None
+    field_value = field.value_from_object(instance) if field else None
+
+    if field_name == 'group':
```
[the remark this answers is not in the document]
Comment (kobayashi) [reply netbox/utilities/utils.py]: If we do get manufacturer from device_type in DeviceForm, we need a query like this.
`kwargs['initial']['manufacturer'] = DeviceType.objects.get(pk=kwargs['initial']['device_type']).manufacturer.pk`
I was concerned this made database access inside form in my first commit. If you think it's ok, I will change to the way.

[the remark this answers is not in the document]
Comment (jeremystretch) [reply netbox/utilities/utils.py]: Yeah, I think that's fine. Maybe not the most efficient approach but definitely the easiest to support at the moment. I would be cautious of overwriting any existing initial data though. Something like:

```python
if 'manufacturer' not in kwargs['initial']:
    device_type_id = kwargs['initial']['device_type']
    # Return the bare slug, or None
    manufacturer_slug = DeviceType.objects.filter(pk=device_type_id).values_list('manufacturer__slug', flat=True).first()
    kwargs['initial']['manufacturer'] = manufacturer_slug
```

[the remark this answers is not in the document]
Comment (kobayashi) [reply netbox/utilities/utils.py]: Modified to get the values in form. Also added test for initial data.

**Pull request merged**